src/port.rs

```diff
@@ -46,12 +46,16 @@ impl Port {
     }
 
     pub unsafe fn set_pin_mode(&mut self, p: usize, mut mode: u32) {
-        let mut pcr = core::ptr::read_volatile(&self.pcr[p]);
+
+        let brw = self.pcr[p];
+        // let mut pcr = core::ptr::read_volatile(&self.pcr[p]);
+        let mut pcr = core::ptr::read_volatile(&brw);
```

Comment:
```suggestion
        let mut pcr = core:ptr::read_volatile(&{brw});
```
We're using brw to `Copy` values before creating the reference, and the goal is just to align the value before use. Instead of creating a new variable for that copy, we can do this by placing the value in a block `{...}` which seems a little cleaner imo.

Reply:
Not sure if/when we should be using raw pointers instead of copying

Reply:
Ah! Maybe cleaner than heavy copy indeed... :)